packages/node/test/parsers.test.ts

```diff
@@ -54,80 +56,74 @@ describe('parsers.ts', () => {
           typeName: 'module.exports../src/index.ts',
         },
       ];
-      return Parsers.parseStack(framesWithFilePath).then(_ => {
-        expect(spy).toHaveBeenCalledTimes(1);
-      });
+      const parsedFrames = Parsers.parseStack(framesWithFilePath);
+      await addContext(parsedFrames);
+      expect(spy).toHaveBeenCalledTimes(1);
     });
 
-    test('parseStack with adding different file', done => {
+    test('parseStack with adding different file', async () => {
       expect.assertions(2);
       let mockCalls = 0;
       let newErrorCalls = 0;
-      void Parsers.parseStack(frames)
-        .then(_ => {
-          mockCalls = spy.mock.calls.length;
-          void Parsers.parseStack(stacktrace.parse(getError()))
-            .then(_1 => {
-              newErrorCalls = spy.mock.calls.length;
-              expect(newErrorCalls).toBeGreaterThan(mockCalls);
-              void Parsers.parseStack(stacktrace.parse(getError()))
-                .then(_2 => {
-                  expect(spy).toHaveBeenCalledTimes(newErrorCalls);
-                  done();
-                })
-                .then(null, () => {
-                  // no-empty
-                });
-            })
-            .then(null, () => {
-              // no-empty
-            });
-        })
-        .then(null, () => {
-          // no-empty
-        });
+      let parsedFrames = Parsers.parseStack(frames);
+      await addContext(parsedFrames);
+
+      mockCalls = spy.mock.calls.length;
+      parsedFrames = Parsers.parseStack(stacktrace.parse(getError()));
+      await addContext(parsedFrames);
+
+      newErrorCalls = spy.mock.calls.length;
+      expect(newErrorCalls).toBeGreaterThan(mockCalls);
+
+      parsedFrames = Parsers.parseStack(stacktrace.parse(getError()));
+      await addContext(parsedFrames);
+
+      expect(spy).toHaveBeenCalledTimes(newErrorCalls);
```

Comment:
Do we need this last bit? Doesn't it just prove what your test above proves, that we're using the cache when possible?